dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java

```diff
@@ -270,29 +272,45 @@ private static void processImageDataForToolVersion(io.dockstore.webservice.core.
         ToolVersion toolVersion) {
         ImageData data = new ImageData();
         List<Checksum> trsChecksums = new ArrayList<>();
-        if (version.getImages() != null) {
+        if (version.getImages() != null && !version.getImages().isEmpty()) {
             version.getImages().forEach(image -> {
                 image.getChecksums().forEach(checksum -> {
                     Checksum trsChecksum = new Checksum();
                     trsChecksum.setType(DOCKER_IMAGE_SHA_TYPE_FOR_TRS);
                     trsChecksum.setChecksum(checksum.getChecksum());
                     trsChecksums.add(trsChecksum);
                 });
+                //TODO: for now, all container images are Docker based
+                data.setImageType(ImageType.DOCKER);
+                data.setSize(image.getSize());
+                data.setUpdated(new Date().toString());
+                data.setImageName(constructName(Arrays.asList(castedContainer.getRegistry(), castedContainer.getNamespace(), castedContainer.getName())));
+                data.setRegistryHost(castedContainer.getRegistry());
+                data.setChecksum(trsChecksums);
+                toolVersion.getImages().add(data);
```

Comment:
Not sure about this but might be something to double-check, data is being added inside the loop but is declared `new` outside the loop. 
Might want to test whether a `version` with multiple images is resulting in a `toolVersion` with multiple images

Reply:
Done